partitioned-heat-conduction/solver-fenics/heat_pySDC_problem_class.py

```diff
@@ -0,0 +1,130 @@
+        # on the non-coupling boundary
+        self.solution_expr = solution_expr
+         
+        # Currently only for Dirichlet boundary, has to be changed for Neumann boundary
+        if self.participant_name == 'Dirichlet':
+            self.couplingBC = df.DirichletBC(self.V, coupling_expr, coupling_boundary)
+            
+        self.remainingBC = df.DirichletBC(self.V, solution_expr, remaining_boundary)
+        
+        # Allow for fixing the boundary conditions for the residual computation
+        # Necessary if imex-1st-order-mass sweeper is used
+        self.fix_bc_for_residual = True
+        
+        # define the homogeneous Dirichlet boundary for residual correction
+        def FullBoundary(x, on_boundary):
+            return on_boundary
+        self.homogenousBC = df.DirichletBC(self.V, df.Constant(0), FullBoundary)
+        
+        
+    def solve_system(self, rhs, factor, u0, t):
+        r"""
+        Solves (M - factor \cdot A) \vec{u} = \vec{rhs}.
+        """
+        u = self.dtype_u(u0)
+        T = self.M - factor * self.K
+        b = self.dtype_u(rhs)
+
+        self.solution_expr.t = t
+        self.remainingBC.apply(T, b.values.vector())
+         
+        if self.participant_name == 'Dirichlet':     
+            # Coupling BC needs to point to correct time
+            dt = t - self.t_start
+            read_data = self.precice.read_data(dt)
+            self.precice.update_coupling_expression(self.coupling_expression, read_data)   
+            
+            self.couplingBC.apply(T, b.values.vector())
+
+        df.solve(T, u.values.vector(), b.values.vector())
+
+        return u
+
+    def eval_f(self, u, t):
+        """
+            Derivative computation.
+            Returns a dtype_f object (rhs_fenics_mesh in our case),
+            with an explicit and implicit part.
+        """
+        f = self.dtype_f(self.V)
+
+        # Implicit part: K*u
+        self.K.mult(u.values.vector(), f.impl.values.vector())
+        
+        # Explicit part: M*g   (g = forcing term)
+        self.forcing_term_expr.t = t
+        f.expl = self.dtype_u(df.interpolate(self.forcing_term_expr, self.V))
+        f.expl = self.apply_mass_matrix(f.expl)
+
+        return f
+    
+    def fix_residual(self, res):
+        """
+            Applies homogeneous Dirichlet boundary conditions to the residual
+        """
+        self.homogenousBC.apply(res.values.vector())
+        return None
+    
+    def apply_mass_matrix(self, u):
+        r"""
+            Returns M*u.
+        """
+
+        me = self.dtype_u(self.V)
+        self.M.mult(u.values.vector(), me.values.vector())
+
+        return me
+
+    def u_exact(self, t):
+        r"""
+            Returns the exact solution at time t.
+        """
+        self.solution_expr.t = t
+        
+        me = self.dtype_u(interpolate(self.solution_expr, self.V), val=self.V)
+        return me
```

Comment:
```suggestion
        return self.dtype_u(interpolate(self.solution_expr, self.V), val=self.V)
```
Would be shorter. Or any reasons for doing it differently?

PS: If this is convention in pySDC. Ok, since you are also doing it similarly in `apply_mass_matrix`